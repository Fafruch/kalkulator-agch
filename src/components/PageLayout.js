/**
 * Created by Fafruch on 18/07/2017.
 */
import React from 'react'
import Header from '../components/Header'
import SubjectsContainer from '../containers/SubjectsContainer'
import Footer from '../components/Footer'

const PageLayout = () => (
  <div>
    <Header />
    <SubjectsContainer subjectsType='Główne' />
    <SubjectsContainer subjectsType='Językowe' />
    <Footer />
  </div>
)

export default PageLayout
