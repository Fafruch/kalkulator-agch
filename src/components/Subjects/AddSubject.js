import React from 'react'
import PropTypes from 'prop-types'

import { PRIMARY_SUBJECTS } from '../../constants/SubjectTypes'

const AddSubject = ({ subjectsType, onAdd }) => (
  <button
    onClick={() => onAdd(subjectsType)}
    className={subjectsType === PRIMARY_SUBJECTS ? 'btn btn-outline-danger' : 'btn btn-outline-success'}
  >
    Dodaj {subjectsType === PRIMARY_SUBJECTS ? 'przedmiot główny' : 'język'}
  </button>
)
AddSubject.propTypes = {
  subjectsType: PropTypes.string.isRequired,
  onAdd: PropTypes.func.isRequired
}

export default AddSubject
