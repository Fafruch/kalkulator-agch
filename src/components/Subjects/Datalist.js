import React from 'react'
import PropTypes from 'prop-types'

import { PRIMARY_SUBJECTS } from '../../constants/SubjectTypes'

const Datalist = ({ subjectsType }) => {
  if (subjectsType === PRIMARY_SUBJECTS) {
    return (
      <datalist id={subjectsType}>
        <option value='Biologia' />
        <option value='Chemia' />
        <option value='Fizyka' />
        <option value='Geografia' />
        <option value='Informatyka' />
        <option value='Matematyka' />

        <option value='Filozofia' />
        <option value='Historia' />
        <option value='Historia sztuki' />
        <option value='Wiedza o społeczeństwie' />
        <option value='Język polski' />
      </datalist>
    )
  }
  return (
    <datalist id={subjectsType}>
      <option value='Język angielski' />
      <option value='Język francuski' />
      <option value='Język hiszpański' />
      <option value='Język niemiecki' />
      <option value='Język rosyjski' />
      <option value='Język włoski' />
    </datalist>
  )
}
Datalist.propTypes = {
  subjectsType: PropTypes.string.isRequired
}

export default Datalist
