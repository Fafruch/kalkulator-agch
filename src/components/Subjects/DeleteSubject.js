import React from 'react'
import PropTypes from 'prop-types'

const DeleteSubject = ({ id, subjectsType, onDelete }) => (
  <img
    src='https://png.icons8.com/delete-bin/ios7/100'
    alt='UsuÅ„ przedmiot'
    onClick={() => onDelete(id, subjectsType)}
    className='subject-icon'
  />
)
DeleteSubject.propTypes = {
  id: PropTypes.string.isRequired,
  subjectsType: PropTypes.string.isRequired,
  onDelete: PropTypes.func.isRequired
}

export default DeleteSubject
