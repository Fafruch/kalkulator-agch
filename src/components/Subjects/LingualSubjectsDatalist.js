import React from 'react'
import PropTypes from 'prop-types'

const LingualSubjectsDatalist = ({ subjectsType }) => (
  <datalist id={subjectsType}>
    <option value='Język angielski' />
    <option value='Język francuski' />
    <option value='Język hiszpański' />
    <option value='Język niemiecki' />
    <option value='Język rosyjski' />
    <option value='Język włoski' />
  </datalist>
)
LingualSubjectsDatalist.propTypes = {
  subjectsType: PropTypes.string.isRequired
}

export default LingualSubjectsDatalist
