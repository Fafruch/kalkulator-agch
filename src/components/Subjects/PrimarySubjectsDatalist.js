import React from 'react'
import PropTypes from 'prop-types'

const PrimarySubjectsDatalist = ({ subjectsType }) => (
  <datalist id={subjectsType}>
    <option value='Biologia' />
    <option value='Chemia' />
    <option value='Fizyka' />
    <option value='Geografia' />
    <option value='Informatyka' />
    <option value='Matematyka' />
  </datalist>
)
PrimarySubjectsDatalist.propTypes = {
  subjectsType: PropTypes.string.isRequired
}

export default PrimarySubjectsDatalist
