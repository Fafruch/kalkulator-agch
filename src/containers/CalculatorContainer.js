import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import Calculator from '../components/Calculator/Calculator'
import subjectsPicker from '../store/selectors/subjectsPicker'
import { toggleScoreTable } from '../actions/index'

const mapStateToProps = (state) => ({
  maxScoreWrapper: subjectsPicker(state.subjects),
  isScoreTableOpened: state.scoreTable.isOpened,
  isAnySubjectPresent: !!state.subjects.primary.length || !!state.subjects.lingual.length
})

const mapDispatchToProps = (dispatch) => ({
  onToggleClick: bindActionCreators(toggleScoreTable, dispatch)
})

const CalculatorContainer = connect(mapStateToProps, mapDispatchToProps)(Calculator)

export default CalculatorContainer
