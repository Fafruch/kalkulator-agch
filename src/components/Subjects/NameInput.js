import React from 'react'
import PropTypes from 'prop-types'

import Datalist from './Datalist'

const NameInput = ({ subject: { id, name, max, active }, subjectsType, onChange }) => (
  <div>
    <input
      type='text'
      value={name}
      list={subjectsType}
      onChange={(event) => onChange(id, 'name', event.target.value, max, subjectsType)}
      className={active ? 'subject-active subject-input-name' : 'subject-not-active subject-input-name'}
    />
    <Datalist subjectsType={subjectsType} />
  </div>
)
NameInput.propTypes = {
  subject: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    elementaryScore: PropTypes.number.isRequired,
    advancedScore: PropTypes.number.isRequired,
    max: PropTypes.object.isRequired,
    active: PropTypes.bool.isRequired
  }).isRequired,
  subjectsType: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}

export default NameInput
