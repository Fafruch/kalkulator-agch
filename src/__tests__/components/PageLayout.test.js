import React from 'react'
import { shallow } from 'enzyme'

import PageLayout from '../../components/PageLayout'
import Subjects from '../../components/Subjects'
import CalculatorContainer from '../../containers/CalculatorContainer'

const setup = (setupProps = {}) => {
  const defaultProps = {}
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<PageLayout {...props} />)

  return {
    props,
    wrapper
  }
}

describe('PageLayout component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
  test('should render two Subjects components', () => {
    const { wrapper } = setup()

    expect(wrapper.find(Subjects)).toHaveLength(2)
  })
  test('should render CalculatorContainer', () => {
    const { wrapper } = setup()

    expect(wrapper.find(CalculatorContainer)).toHaveLength(1)
  })
})
