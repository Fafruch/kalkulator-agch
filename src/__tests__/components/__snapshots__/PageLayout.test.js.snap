// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container"
>
          <Header />
          <Introduction />
          <Subjects
                    subjectsType="primary"
          />
          <Subjects
                    subjectsType="lingual"
          />
          <Connect(Calculator) />
</div>,
        <div>
          <Footer />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header />,
            <Introduction />,
            <Subjects
              subjectsType="primary"
/>,
            <Subjects
              subjectsType="lingual"
/>,
            <Connect(Calculator) />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "subjectsType": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "subjectsType": "lingual",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Footer />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container"
>
            <Header />
            <Introduction />
            <Subjects
                        subjectsType="primary"
            />
            <Subjects
                        subjectsType="lingual"
            />
            <Connect(Calculator) />
</div>,
          <div>
            <Footer />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header />,
              <Introduction />,
              <Subjects
                subjectsType="primary"
/>,
              <Subjects
                subjectsType="lingual"
/>,
              <Connect(Calculator) />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "subjectsType": "primary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "subjectsType": "lingual",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Footer />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
